app_flutter/lib/state/flutter_build.dart

```diff
@@ -50,14 +56,22 @@ class FlutterBuildState extends ChangeNotifier {
         Timer.periodic(refreshRate, (_) => _fetchBuildStatusUpdate());
   }
 
-  /// Request the latest [statuses] from [CocoonService].
+  /// Request the latest [statuses] and [isTreeBuilding] from [CocoonService].
   Future<void> _fetchBuildStatusUpdate() async {
-    await Future.wait<void>(<Future<void>>[
-      _cocoonService.fetchCommitStatuses().then<List<CommitStatus>>(
-          (List<CommitStatus> commitStatuses) => statuses = commitStatuses),
-      _cocoonService
-          .fetchTreeBuildStatus()
-          .then<bool>((bool treeStatus) => _isTreeBuilding = treeStatus),
+    await Future.wait(<Future<void>>[
+      _cocoonService.fetchCommitStatuses().then(
```

Comment:
Should the _cocoonServer be responsible for determining what is and isn't an error?

Reply:
Each implementation of CocoonService has to decide what is an error and what isn't an error. I don't think that can be done in FlutterBuildState where it can take in different implementations of CocoonService. This might lead to over generalizing what an error is.

Since the logic behind handling errors is the same, I think we should handle this in FlutterBuildState to just write this error handling code once instead of in each CocoonService implementation.

Reply:
From the example code here, the _cocoonService is already deciding what is and isn't an error by throwing vs returning normally.

Reply:
Yes, the CocoonService should determine what is an isn't an error so FlutterBuildState doesn't need any extra logic handling errors.

Reply:
Why not return a CocoonResponse object instead?

Reply:
Done.

This makes more sense since it simplifies FlutterBuildState to just hold data. No extra responsibilities :)

Thanks for the suggestion!